example/hr-data-generator/src/main/java/uk/gov/gchq/palisade/example/hrdatagenerator/types/Employee.java

```diff
@@ -47,6 +46,64 @@
     private WorkLocation workLocation;
     private Sex sex;
 
+    public Employee() {
+    }
+
+    public Employee(final Employee employee) {
+        name = employee.name;
+        dateOfBirth = employee.dateOfBirth;
+        taxCode = employee.taxCode;
+        nationality = employee.nationality;
+        hireDate = employee.hireDate;
+        grade = employee.grade;
+        department = employee.department;
+        salaryAmount = employee.salaryAmount;
+        salaryBonus = employee.salaryBonus;
+        sex = employee.sex;
+
+        if (employee.uid != null) {
+            uid = employee.uid.clone();
```

Comment:
in an ideal world the UserId should also use the copy clone pattern. However given we are mid transition and the common repo has already been moved which uses the clone method. I suggest we leave the UserId as is and make a not to change it in the Palisade-common repo instead